orrery/src/executor.cpp

```diff
@@ -0,0 +1,52 @@
+// Concord
+//
+// Copyright (c) 2021 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the
+// "License").  You may not use this product except in compliance with the
+// Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the subcomponent's license, as noted in the LICENSE
+// file.
+
+#include "orrery/executor.h"
+#include <variant>
+#include "assertUtils.hpp"
+#include "orrery_msgs.cmf.hpp"
+
+using namespace std::chrono_literals;
+
+namespace concord::orrery {
+
+static constexpr std::chrono::milliseconds TIMEOUT = 100ms;
+
+std::thread Executor::start() && {
+  return std::thread([executor{std::move(*this)}]() mutable {
```

Comment:
Would it make sense to call `executor.join()` instead in main(), avoiding the moved-from executor? And having the thread as a member of `Executor`.

Reply:
Yes, I thought about this afterwards as well. I believe this is the correct thing to do. This pattern is a bit confusing. I will leave this as follow up to whoever takes ownership of orrery 😛